src/components/shared/ContactHostForm/ContactHostForm.tsx

```diff
@@ -0,0 +1,140 @@
+  message: string;
+  recipientId: string;
+  subject: string;
+}
+
+const ContactHostSchema = Yup.object().shape({
+  subject: Yup.string().required('Please fill out the subject.'),
+  message: Yup.string().required('Please fill out the message field.'),
+});
+
+const ContactHostForm = (props: Props) => {
+  const { host, listingId, onClose, tripId } = props;
+    return (
+      <ContactHostFormContainer>
+        <Formik
+          initialValues={{
+            subject: '',
+            message: '',
+          }}
+          isInitialValid
+          validationSchema={ContactHostSchema}
+          onSubmit={(values: ContactHostInput, actions) => {
+          actions.setSubmitting(true);
+          const input = {
+            ...values,
+            bookingId: tripId,
+            listingId,
+            recipientId: host.id,
+          };
+          return props.contactUser(input)
+            .then((returnedObject: EmailResponse) => {
+              console.log('returnedObject:', returnedObject);
+            })
+            .catch((error: Error) => {
+              alert(`${error}. If this continues to occur, please contact us at support@beetoken.com`);
+              console.error(error);
+              return actions.setSubmitting(false);
+            });
+          }}
+        >
+          {({
+            isSubmitting,
+            setFieldTouched,
+            setFieldValue,
+            values,
+          }) => (
+            <Form>
+              <h2>Contact {host.firstName || 'Host'}</h2>
+              
+              <div className="form-item">
+                <InputLabel htmlFor="subject">Subject</InputLabel>
+                <InputWrapper>
+                  <Field
+                    name="subject"
+                    placeholder="Subject"
+                    type="text" />
+                </InputWrapper>
+                <ErrorMessageWrapper>
+                  <ErrorMessage name="subject" />
+                </ErrorMessageWrapper>
+              </div>
+
+              <div className="form-item">
+                <InputLabel>Message</InputLabel>
+                <Textarea
+                  html
+                  name="message"
+                  onBlur={() => setFieldTouched('message', true)}
+                  onChange={(event: TextareaEvent) => {
+                    setFieldValue('message', event.target.value);
+                  }}
+                  placeholder="Let the host know of any questions or concerns you may have"
+                  textareaHeight="200px"
+                  value={values.message} />
+                <ErrorMessageWrapper>
+                  <ErrorMessage name="message" />
+                </ErrorMessageWrapper>
+              </div>
+
+              <Button
+                disabled={isSubmitting}
+                type="submit">
+                Send Message
+              </Button>
+            </Form>
+          )}
+        </Formik>
+        <div className="close" onClick={onClose}>
```

Comment:
Dont we have a shared `close` component?

Reply:
ya sry. newest code push should have <Closebutton />